typescript/infra/config/environments/mainnet3/agent.ts

```diff
@@ -344,101 +334,49 @@ const gasPaymentEnforcement: GasPaymentEnforcement[] = [
-  // users have the best possible experience, we whitelist messages between
-  // warp routes that we know are certainly paying for gas.
-  {
-    type: GasPaymentEnforcementPolicyType.None,
-    matchingList: [
-      ...routerMatchingList(injectiveInevmInjAddresses),
-      // As we are still indexing the IGP on Stride, temporarily whitelist
-      // Stride to Eclipse messages.
-      ...routerMatchingList(eclipseStrideTiaAddresses),
-      ...routerMatchingList(eclipseStrideStTiaAddresses),
-    ],
-  },
   {
     type: GasPaymentEnforcementPolicyType.OnChainFeeQuoting,
   },
 ];
 
-const metricAppContexts = [
-  {
-    name: 'helloworld',
-    matchingList: routerMatchingList(helloWorld[Contexts.Hyperlane].addresses),
-  },
-  {
-    name: 'injective_inevm_inj',
-    matchingList: routerMatchingList(injectiveInevmInjAddresses),
-  },
-  {
-    name: 'inevm_ethereum_usdc',
-    matchingList: matchingList(inevmEthereumUsdcAddresses),
-  },
-  {
-    name: 'inevm_ethereum_usdt',
-    matchingList: matchingList(inevmEthereumUsdtAddresses),
-  },
-  {
-    name: 'viction_ethereum_eth',
-    matchingList: routerMatchingList(victionEthereumEthAddresses),
-  },
-  {
-    name: 'viction_ethereum_usdc',
-    matchingList: routerMatchingList(victionEthereumUsdcAddresses),
-  },
-  {
-    name: 'viction_ethereum_usdt',
-    matchingList: routerMatchingList(victionEthereumUsdtAddresses),
-  },
-  {
-    name: 'ancient8_ethereum_usdc',
-    matchingList: routerMatchingList(ancient8EthereumUsdcAddresses),
-  },
-  {
-    name: 'renzo_ezeth',
-    matchingList: matchingList(renzoEzEthAddressesV3),
-  },
-  {
-    name: 'eclipse_usdc',
-    matchingList: matchingList(
-      getWarpAddresses(WarpRouteIds.EclipseEthereumSolanaUSDC),
-    ),
-  },
-  {
-    name: 'eclipse_teth',
-    matchingList: matchingList(
-      getWarpAddresses(WarpRouteIds.EclipseEthereumTETH),
-    ),
-  },
-  {
-    name: 'eclipse_wif',
-    matchingList: matchingList(getWarpAddresses(WarpRouteIds.EclipseSolanaWIF)),
-  },
-  {
-    name: 'eclipse_sol',
-    matchingList: matchingList(getWarpAddresses(WarpRouteIds.EclipseSolanaSOL)),
-  },
-  // Hitting max env var size limits, see https://stackoverflow.com/questions/28865473/setting-environment-variable-to-a-large-value-argument-list-too-long#answer-28865503
-  // {
-  //   name: 'merkly_erc20',
-  //   matchingList: routerMatchingList(merklyErc20Addresses),
-  // },
-  // {
-  //   name: 'merkly_eth',
-  //   matchingList: routerMatchingList(merklyErc20Addresses),
-  // },
-  // {
-  //   name: 'merkly_nft',
-  //   matchingList: routerMatchingList(merklyErc20Addresses),
-  // },
-];
+// Gets metric app contexts, including:
+// - helloworld
+// - all warp routes defined in WarpRouteIds, using addresses from the registry
+// - misc important applications not defined in the registry, e.g. merkly
+const metricAppContextsGetter = (): MetricAppContext[] => {
+  const warpContexts = Object.values(WarpRouteIds).map((warpRouteId) => {
```

Comment:
whats the motivation behind a subset of the registry, so that our metrics dont get polluted?

Reply:
yeah exactly - atm we have high urgency alerts for these if the prep queue goes up, for now I'd like us to be stringent about which apps become eligible for a high urgency alert. Def open to changing this in the future though, ideally when it's easier to configure alerts via iaac or something